app/controllers/parent_objects_controller.rb

```diff
@@ -59,8 +60,21 @@ def valid_request?
 
   # PATCH/PUT /parent_objects/1
   # PATCH/PUT /parent_objects/1.json
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/PerceivedComplexity
   def update
     respond_to do |format|
+      parent_object = ParentObject.find(params[:id])
+      permission_set = parent_object&.permission_set
+
+      authorize!(:owp_access, permission_set) if parent_object.visibility == "Open with Permission" && parent_object.visibility != parent_object_params[:visibility]
+
+      authorize!(:owp_access, permission_set) if permission_set.present? && parent_object_params[:permission_set_id].nil?
+
+      authorize!(:owp_access, OpenWithPermission::PermissionSet) if permission_set.nil? && parent_object_params[:permission_set_id].present?
```

Comment:
Does this confirm that the user has access to the PermissionSet identified by  `parent_object_params[:permission_set_id]` ?  I think that's the desired case if someone is moving a Parent out of a PermissionSet.

Reply:
You should remove `permission_set.nil?` from the if too. It should check the specific PermissionSet the user is setting it to regardless of `permission_set.nil?`.